File: RequestsMicroservice/src/RequestsMicroservice/Database/Batch.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace RequestsMicroservice.Database
{
    public class Batch
    {
        [Key]
        public Guid BatchId { get; set; }

        [Required]
        [MaxLength(64)]
        public string Description { get; set; }

        [Required]
        public DateTime Created { get; set; }

        #region Dependencies

        public ICollection<Request> Requests { get; set; }

        #endregion
    }
}

File: UI/src/microservices/requests/Batch.ts
export interface IBatch {
  readonly batchId: string;
  readonly description: string;
  readonly created: Date;
}

export class Batch implements IBatch {
  readonly batchId: string;
  readonly description: string;
  readonly created: Date;

  constructor(data: { batchId: string, description: string, created: any}) {
    this.batchId = data.batchId;
    this.description = data.description;
    this.created = data.created;
  }
}
